### DISCUSSION TITLE
Number of downloads of beta.8 for the platform packages

### DISCUSSION THREAD

**Discussion by pauldendulk** [Mapsui/Mapsui#1866] [Development]
To get an impression of what is used the most I added the number of download of beta.8 that this table I created by hand.

MAUI is clearly used the most. Avalonia (which was released later) already surpassed Xamarin.Forms. I am surprised by how high WinUI is. Perhaps this is WPF users migrating (we always had a fair amount of WPF users). Blazor is rising and I expect this trend to continue the coming releases. Uno could be higher if more users transition to Uno.WinUI.

These numbers are just an impression and could be skewed for a number of reasons. For instance, right after the release there are a lot of immediate downloads of all packages, which I attribute to some bots analyzing stuff. 

| Package | Version  | Downloads | Quickstart Guide | Downloads |
| ---------------|:-------------:|:-------------:|:-------------:|:------------------------------------------------------------------------------------:|
| Mapsui.Maui   | [![NuGet Status](https://img.shields.io/nuget/v/Mapsui.Maui.svg?style=flat)](https://www.nuget.org/packages/Mapsui.Maui/) | [![NuGet](https://img.shields.io/nuget/dt/Mapsui.Maui.svg)](https://www.nuget.org/packages/Mapsui.Maui) |          [MAUI](https://mapsui.com/documentation/getting-started-maui.html)          | 219 |
| Mapsui.Wpf     | [![NuGet Status](https://img.shields.io/nuget/v/Mapsui.Wpf.svg?style=flat)](https://www.nuget.org/packages/Mapsui.Wpf/) |[![NuGet](https://img.shields.io/nuget/dt/Mapsui.Wpf.svg)](https://www.nuget.org/packages/Mapsui.Wpf) |           [WPF](https://mapsui.com/documentation/getting-started-wpf.html)           | 74 |
| Mapsui.Avalonia     | [![NuGet Status](https://img.shields.io/nuget/v/Mapsui.Avalonia.svg?style=flat)](https://www.nuget.org/packages/Mapsui.Avalonia/) |[![NuGet](https://img.shields.io/nuget/dt/Mapsui.Avalonia.svg)](https://www.nuget.org/packages/Mapsui.Avalonia) |      [Avalonia](https://mapsui.com/documentation/getting-started-avalonia.html)      | 63 |
| Mapsui.Forms   | [![NuGet Status](https://img.shields.io/nuget/v/Mapsui.Forms.svg?style=flat)](https://www.nuget.org/packages/Mapsui.Forms/) | [![NuGet](https://img.shields.io/nuget/dt/Mapsui.Forms.svg)](https://www.nuget.org/packages/Mapsui.Forms) | [Xamarin Forms](https://mapsui.com/documentation/getting-started-xamarin-forms.html) | 60 |
| Mapsui.WinUI     | [![NuGet Status](https://img.shields.io/nuget/v/Mapsui.WinUI.svg?style=flat)](https://www.nuget.org/packages/Mapsui.WinUI/) |[![NuGet](https://img.shields.io/nuget/dt/Mapsui.WinUI.svg)](https://www.nuget.org/packages/Mapsui.WinUI) |         [WinUI](https://mapsui.com/documentation/getting-started-winui.html)         | 56 |
| Mapsui.Blazor     | [![NuGet Status](https://img.shields.io/nuget/v/Mapsui.Blazor.svg?style=flat)](https://www.nuget.org/packages/Mapsui.Blazor/) |[![NuGet](https://img.shields.io/nuget/dt/Mapsui.Blazor.svg)](https://www.nuget.org/packages/Mapsui.Blazor) |        [Blazor](https://mapsui.com/documentation/getting-started-blazor.html)        | 53 |
| Mapsui.Uno.WinUI     | [![NuGet Status](https://img.shields.io/nuget/v/Mapsui.Uno.WinUI.svg?style=flat)](https://www.nuget.org/packages/Mapsui.Uno.WinUI/) |[![NuGet](https://img.shields.io/nuget/dt/Mapsui.Uno.WinUI.svg)](https://www.nuget.org/packages/Mapsui.Uno.WinUI) |     [Uno.WinUI](https://mapsui.com/documentation/getting-started-uno-winui.html)     | 47 |
| Mapsui.Android | [![NuGet Status](https://img.shields.io/nuget/v/Mapsui.Android.svg?style=flat)](https://www.nuget.org/packages/Mapsui.Android/) |[![NuGet](https://img.shields.io/nuget/dt/Mapsui.Android.svg)](https://www.nuget.org/packages/Mapsui.Android) |       [Xamarin Android](https://mapsui.com/documentation/getting-started-android.html)       | 47 |
| Mapsui.Uno     | [![NuGet Status](https://img.shields.io/nuget/v/Mapsui.Uno.svg?style=flat)](https://www.nuget.org/packages/Mapsui.Uno/) |[![NuGet](https://img.shields.io/nuget/dt/Mapsui.Uno.svg)](https://www.nuget.org/packages/Mapsui.Uno) |           [Uno](https://mapsui.com/documentation/getting-started-uno.html)           | 45 |
| Mapsui.iOS     | [![NuGet Status](https://img.shields.io/nuget/v/Mapsui.iOS.svg?style=flat)](https://www.nuget.org/packages/Mapsui.iOS/) |[![NuGet](https://img.shields.io/nuget/dt/Mapsui.iOS.svg)](https://www.nuget.org/packages/Mapsui.iOS) |           [Xamarin iOS](https://mapsui.com/documentation/getting-started-ios.html)           | 41 |
| Mapsui.Eto     | [![NuGet Status](https://img.shields.io/nuget/v/Mapsui.Eto.svg?style=flat)](https://www.nuget.org/packages/Mapsui.Eto/) |[![NuGet](https://img.shields.io/nuget/dt/Mapsui.Eto.svg)](https://www.nuget.org/packages/Mapsui.Eto) |           [Eto](https://mapsui.com/documentation/getting-started-eto.html)           | 40 |

**Comment by IldarKhayrutdinov** [comment]
I migrated to `Avalonia` because of this problem: https://github.com/microsoft/microsoft-ui-xaml/issues/5917

**Comment by inforithmics** [comment]
Yes the usage of Uno.WinUI could increase because there is now an offical Sample with Mapsui as Mapping Control

https://github.com/unoplatform/Uno.Samples/tree/master/UI/MapControlSample

**Comment by p2pbsh** [comment]
I'm trying to test out the Blazor package in beta 8 according to the getting started guide, but the build is failing due to .Map in _mapControl.Map being an unknown type:

`_mapControl.Map?.Layers.Add(Mapsui.Tiling.OpenStreetMap.CreateTileLayer());`

Kinda makes sense as _mapControl is an ElementReference object as opposed to a concrete MapControl.

Am I being stupid or is the documentation not quite right?

**Comment by inforithmics** [comment]
I'll update the documentation

Following Code should work _mapControl had the wrong type should have been MapControl instead of ElementReference

```
<div class="container">
    <div class="row">
        <div class="col border rounded p-2 canvas-container">
            <MapControlComponent @ref="_mapControl" />
        </div>
    </div>
</div>

@code 
{
    private MapControl? _mapControl;

    protected override void OnAfterRender(bool firstRender)
    {
        base.OnAfterRender(firstRender);
        if (firstRender)
        {
            if (_mapControl != null)
                _mapControl.Map?.Layers.Add(Mapsui.Tiling.OpenStreetMap.CreateTileLayer());
        }
    }
}
```

**Comment by p2pbsh** [comment]
Thanks for the assist, sorted.

I was indeed being stupid, I tried that previously but wasn't pulling in the correct namespace, would recommend adding to the docs too:

`@using Mapsui.UI.Blazor;`

Btw, map mouse wheel zooming seems to be back to front :)

**Reply by pauldendulk** [reply]
> Btw, map mouse wheel zooming seems to be back to front :)

@inforithmics I noticed this too. It is inverted compared to all other platforms? Can this be easily fixed or is this there a Mac versus Windows complication?

**Reply by p2pbsh** [reply]
Can confirm it's incorrect viewing on Windows, if that helps. Default behaviour is definitely reversed compared to the UWP and MAUI implementations.

**Comment by inforithmics** [comment]
Pull request for fixing ScrollWheel Zooming

https://github.com/Mapsui/Mapsui/pull/1881

**Comment by djjeffg382** [comment]
I'm giving the Blazor target a try.   I can't figure how to get access to resize the canvas larger than the default.  Is this possible?

**Reply by pauldendulk** [reply]
Could you post this as a new question https://github.com/Mapsui/Mapsui/discussions? And are you trying to resize the MapControl itself, or the part of the map that you see in the MapControl?
